File: Example/Example for lottie-macos/ViewController.m
//
//  ViewController.m
//  Example for lottie-macos
//
//  Created by Oleksii Pavlovskyi on 2/2/17.
//  Copyright © 2017 Brandon Withrow. All rights reserved.
//

#import "ViewController.h"
#import <Lottie/Lottie.h>
#import "LAMainView.h"
#import "LottieFilesURL.h"

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
}

- (void)viewDidAppear {
  [super viewDidAppear];
}

- (void)viewDidDisappear {
  [super viewDidDisappear];
}

- (IBAction)_sliderChanged:(NSSlider *)sender {
  [(LAMainView *)self.view setAnimationProgress:sender.floatValue];
}

- (IBAction)_rewind:(id)sender {
  [(LAMainView *)self.view rewindAnimation];
}

- (IBAction)_play:(id)sender {
  [(LAMainView *)self.view playAnimation];
}

- (IBAction)_loops:(id)sender {
  [(LAMainView *)self.view toggleLoop];
}

- (void)paste:(id)sender {
    NSPasteboard *pasteboard = [NSPasteboard generalPasteboard];
    NSArray *classes = [[NSArray alloc] initWithObjects:[NSURL class], nil];
    
    if ([pasteboard canReadObjectForClasses:classes options:nil]) {
        NSArray *copiedItems = [pasteboard readObjectsForClasses:classes options:nil];
        
        if (copiedItems != nil) {
            NSURL *url = (NSURL *)[copiedItems firstObject];
            LottieFilesURL *lottieFile = [[LottieFilesURL alloc] initWithURL:url];
            
            if (lottieFile != nil) {
                [(LAMainView *)self.view openAnimationURL:lottieFile.jsonURL];
                self.view.window.title =  lottieFile.animationName;
            }
        }
    }
}


@end

File: Example/macOS/ViewController.swift
//
//  ViewController.swift
//  lottie-swift_macOS
//
//  Created by Brandon Withrow on 2/5/19.
//  Copyright © 2019 CocoaPods. All rights reserved.
//

import Cocoa
import Lottie

class ViewController: NSViewController {

  override func viewDidLoad() {
    super.viewDidLoad()

    let animation = LottieAnimation.named("Samples/LottieLogo1")
    let animationView = LottieAnimationView(animation: animation)
    view.addSubview(animationView)
    preferredContentSize = animationView.bounds.size
    animationView.loopMode = .loop
    animationView.play()
  }

}
